domain/application/service/application.go

```diff
@@ -69,15 +77,17 @@ func NewApplicationService(st ApplicationState, registry storage.ProviderRegistr
 	}
 }
 
-// CreateApplication creates the specified application and units if required.
+// CreateApplication creates the specified application and units if required,
+// returning an error satisfying [applicationerrors.ApplicationAle\readyExists]
+// if the application already exists.
 func (s *ApplicationService) CreateApplication(
 	ctx context.Context,
-	name string, charm charm.Charm,
```

Comment:
I did this on purpose. This is because args implies that you can have optional things. Charm is definitely not optional when creating an application. Can we put this back.

Reply:
Added origin as a positional param as it's required. Also added validation check which exposed deficiencies in existing unit tests.